web/repositories_test.go

```diff
@@ -191,3 +192,120 @@ func TestGetRepositories(t *testing.T) {
 		}
 	}
 }
+
+func TestQuickFeedService_isEmptyRepo(t *testing.T) {
+	db, cleanup := qtest.TestDB(t)
+	defer cleanup()
+	client := web.MockClient(t, db, nil)
+
+	user := qtest.CreateFakeUser(t, db)
+	course := qtest.MockCourses[0]
+	qtest.CreateCourse(t, db, user, course)
+
+	student := qtest.CreateFakeUser(t, db)
+	qtest.EnrollStudent(t, db, student, course)
+
+	// student, in a group
+	groupStudent := qtest.CreateFakeUser(t, db)
+	qtest.EnrollStudent(t, db, groupStudent, course)
+
+	group := &qf.Group{
+		Name:     "1001 Hacking Crew",
+		CourseID: course.ID,
+		Users:    []*qf.User{groupStudent},
+	}
+	if err := db.CreateGroup(group); err != nil {
+		t.Fatal(err)
+	}
+
+	// create repositories for users and group
+	userRepo := &qf.Repository{
+		ScmOrganizationID: course.ScmOrganizationID,
+		ScmRepositoryID:   1,
+		UserID:            user.ID, // 1
+		HTMLURL:           "user",
+		RepoType:          qf.Repository_USER,
+	}
+	if err := db.CreateRepository(userRepo); err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name    string
+		ctx     context.Context
+		request *qf.RepositoryRequest
+		wantErr bool
+	}{
+		{
+			// cannot distinguish between empty and non-existing repositories
+			name: "empty repositories",
+			ctx:  context.Background(),
+			request: &qf.RepositoryRequest{
+				CourseID: course.ID, // 1
+				GroupID:  group.ID,  // 1
+			},
+			wantErr: false,
+		},
+		{
+			name: "no repositories",
+			ctx:  context.Background(),
+			request: &qf.RepositoryRequest{
+				CourseID: course.ID,
+				GroupID:  group.ID, // 1
+			},
+			wantErr: false,
+		},
+		{
+			name: "course not found",
+			ctx:  context.Background(),
+			request: &qf.RepositoryRequest{
+				CourseID: 123,
+				UserID:   user.ID, // 1
+			},
+			// unable to get SCM client for unknown course -> error
+			wantErr: true,
+		},
+		{
+			name: "user not found",
+			ctx:  context.Background(),
+			request: &qf.RepositoryRequest{
+				CourseID: course.ID, // 1
+				UserID:   123,
+			},
+			// lookup for invalid user should return no repositories
+			wantErr: false,
+		},
+		{
+			name: "user has no repositories",
+			ctx:  context.Background(),
```

Comment:
Can avoid these in the struct, and just use `context.Background()` directly in the call.